python/CSXCAD/CSRectGrid.pyx

```diff
@@ -167,11 +163,10 @@ cdef class CSRectGrid:
         """
         self.thisptr.clear()
 
-    def SetDeltaUnit(self, unit):
-        """ SetDeltaUnit(unit)
-
-        Set the drawing unit for all mesh lines. Default is 1 (m)
-        """
+    def SetDeltaUnit(self, unit:float=1):
+        """Set the drawing unit for all mesh lines, e.g. `unit=1` means units will be in meter, `unit=1e-3` means units will be in millimeter, and so."""
+        if not isinstance(unit, (int,float)) or not unit>0:
```

Comment:
Again this is unnecessary and potentially to restrictive because e.g. a np.float would fail for no reason and any invalid value would cause an exception anyways

Reply:
For me it works fine (OS = Fedora 40, Python and NumPy versions below):
```python
$ ipython
Python 3.12.7 (main, Oct  1 2024, 00:00:00) [GCC 14.2.1 20240912 (Red Hat 14.2.1-3)]
Type 'copyright', 'credits' or 'license' for more information
IPython 8.23.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import numpy

In [2]: n = numpy.array([1.1])[0]

In [3]: type(n), isinstance(n, float)
Out[3]: (numpy.float64, True)

In [4]: numpy.__version__
Out[4]: '1.26.4'
```

Reply:
Well than this was a problem in older python. In any case, what does this check solve here?
If the input is invalid, it will fail with an error one line later anyhow?

Reply:
In my experience it is usually the best to have [a fail-fast approach](https://en.wikipedia.org/wiki/Fail-fast_system), so it becomes trivial for the user to track down any mistake, and it is never bad to provide explicit error messages as in this case. If it fails later on, especially inside the C++ core, the error message could be cryptic for a non experienced user, or completely inexistent, as in the case of the issue I opened about `port_nr`, and you know about it just because (hopefully) you notice the results don't make sense. I don't think that having this check here creates any issue.

About the way the type is checked, NumPy does the same, `isinstance(n, (float,int))`. An example [here](https://github.com/numpy/numpy/blob/2f7fe64b8b6d7591dd208942f1cc74473d5db4cb/numpy/_core/function_base.py#L303C23-L303C27). I quote NumPy just because it is a well established package used by millions.

Reply:
`mesh.SetDeltaUnit('fails')

File "CSXCAD/CSRectGrid.pyx", line 175, in CSXCAD.CSRectGrid.CSRectGrid.SetDeltaUnit
TypeError: must be real number, not str`

The check itself is fine then, but sorry this message is plenty clear and fails at exactly the same line. Duplicating checks is not good and makes it not fail any faster...

Reply:
What about `mesh.SetDeltaUnit(-1)`? For me this produces:
```
$ python run.py 
 ---------------------------------------------------------------------- 
 | openEMS 64bit -- version v0.0.36-20-gceb80b5
 | (C) 2010-2023 Thorsten Liebig <thorsten.liebig@gmx.de>  GPL license
 ---------------------------------------------------------------------- 
	Used external libraries:
		CSXCAD -- Version: v0.6.3-12-g25159c0
		hdf5   -- Version: 1.12.1
		          compiled against: HDF5 library version: 1.12.1
		tinyxml -- compiled against: 2.6.2
		fparser
		boost  -- compiled against: 1_83
		vtk -- Version: 9.2.6
		       compiled against: 9.2.6

Create FDTD operator (compressed SSE + multi-threading)
CartOperator::SetupCSXGrid: grid delta unit must not be <=0 !!!
Traceback (most recent call last):
  File "/home/myself/testing/run.py", line 100, in <module>
    run_simulation(
  File "/home/myself/testing/run.py", line 90, in run_simulation
    simulation.Run(
  File "openEMS/openEMS.pyx", line 571, in openEMS.openEMS.openEMS.Run
RuntimeError: Setup failed, error code: 2
```
As a Python user, I look at the Python error stack trace: 
-  `error code 2` is telling me nothing
-  It is failing in the `openEMS.Run` method, which is several lines away from where the mistake actually is. 

There is the extra information `CartOperator::SetupCSXGrid: grid delta unit must not be <=0 !!!` which is a hint, and in a trivial code like the one I have now it is enough, but in a more complex setup you may have no idea where this is coming from. Having the error stack trace pointing to the mistake is the most user friendly.

`SetDeltaUnit(unit=float('NaN'))` also produces a cryptic error. No human will ever write this, but it may come if you somehow calculate the unit automatically.

Reply:
I find the message `CartOperator::SetupCSXGrid: grid delta unit must not be <=0 !!!`  very very clear!!
And this is the correct place to check and warn about this... This is not the task of the wrapper... because this could happen via Octave or C++ interface too..  Additionally this is CSXCAD and not openEMS, maybe some other solver wants or even needs to have this negative? I really don't think so, but who knows...